### PR TITLE
TEST-3378: Restruct the stateful directory layout

### LINKED ISSUE #3378: Re-struct suites/0_stateful directory layout
**Summary**

Now:
```
0_stateful/
├── 12_0000_copy_s3.result
├── 12_0000_copy_s3.sql
├── 12_0001_streaming_load.result
├── 12_0001_streaming_load.sh
├── ddl
└── sql
```

expect:
```
0_stateful/
├── ontime/
├── copy/
├── data_loading/

```

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR:
```
0_stateful/
├── 01_ontime/
├── 02_data_load/
```

## Changelog

- Improvement

## Related Issues

Fixes #3378

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergifyio update

File: scripts/ci/ci-run-stateful-tests-cluster.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ fi
 echo "Starting stateful databend-test"
 # Create table
 # shellcheck disable=SC2044
-for i in $(find ./tests/suites/0_stateful/sql -name "*.sql" -type f); do
+for i in $(find ./tests/suites/0_stateful/01_ontime/sql -name "*.sql" -type f); do
```
Comment (sundy-li) [inline scripts/ci/ci-run-stateful-tests-cluster.sh:50]: Seems this script is duplicated with `databend-test`.

For stateful tests:

Step 1:  Load DDL SQL queries to create tables.
Step 2:  Load sample data into the tables.
Step 3: use `./databend-test --run-dir 0_stateful`

Comment (BohuTANG) [reply scripts/ci/ci-run-stateful-tests-cluster.sh:50]: Ok.

This PR try to re-fine the stateful test directory layout, to make it more readable.
For example:
01_ontime -- this is a ontime dataset test.

I am not sure the ontime dataset willbe used by other stateful tests.

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
Stateful(Standalone) and Stateless(Cluster)  tests all is ok in my local PC(ubuntu), not sure why here is failed :thinking:

File: scripts/ci/ci-run-stateful-tests-cluster.sh (hunk 1 above)
Comment (BohuTANG) [reply scripts/ci/ci-run-stateful-tests-cluster.sh:50]: Change it to:
```
SCRIPT_PATH="$(cd "$(dirname "$0")" >/dev/null 2>&1 && pwd)"
cd "$SCRIPT_PATH/../../tests" || exit

echo "Starting databend-test"
./databend-test --mode 'cluster' --run-dir 0_stateful
```

but got:
```
successfull loaded ontime dataset
Starting stateful databend-test
No tests were run.
make: *** [Makefile:119: stateful-cluster-test] Error 1
```

**Comment by sundy-li** [comment]
Continue #3488

**Pull request closed without merging**